sysfs/vulnerability.go

```diff
@@ -59,29 +73,37 @@ func (fs FS) CPUVulnerabilities() ([]Vulnerability, error) {
 // Vulnerability represents a single vulnerability extracted from /sys/devices/system/cpu/vulnerabilities/.
 type Vulnerability struct {
 	CodeName   string
-	State      string
+	State      int
 	Mitigation string
 }
 
-func parseVulnerability(name, value string) (Vulnerability, error) {
-	v := Vulnerability{CodeName: name}
-	value = strings.TrimSpace(value)
-	if value == notAffected {
-		v.State = notAffected
+func parseVulnerability(name, rawContent string) (*Vulnerability, error) {
+	v := &Vulnerability{CodeName: name}
+	rawContent = strings.TrimSpace(rawContent)
+	rawContentLower := strings.ToLower(rawContent)
+
+	if strings.HasPrefix(rawContentLower, notAffected) {
```

Comment:
Turn these into switch cases

Reply:
Also why HasPrefix and not ==?

Reply:
> Turn these into switch cases

yup, makes sense, will push a change in a sec

> Also why HasPrefix and not ==?

that's because the actual content returned from sysfs can have multiple different formats, e.g.:
- `Vulnerable`
- `Vulnerable: Clear CPU buffers attempted, no microcode`
- `Mitigation`
- `Mitigation: Retpolines, IBPB: conditional, STIBP: always-on, RSB filling, PBRSB-eIBRS: Not affected`
- `Not affected`

(see fixtures for more examples)

And checking the prefix was one of the simplest solutions I could come up with for identifying the state. I don't really like this approach, I'm open to any suggestion on how this could be done better